nearcore/src/lib.rs

```diff
@@ -110,20 +120,22 @@ fn open_storage(home_dir: &Path, near_config: &NearConfig) -> anyhow::Result<Nod
         res.with_context(|| format!("unable to open database at {}", opener.path().display()))?;
 
     // Check if the storage is an archive and if it is make sure we are too.
-    // If the store is not marked as archive but we are an archival node that is
-    // fine and we just need to mark the store as archival.
     let hot = storage.get_store(Temperature::Hot);
     let store_is_archive: bool =
         hot.get_ser(DBCol::BlockMisc, near_store::db::IS_ARCHIVE_KEY)?.unwrap_or_default();
-    let client_is_archive = near_config.client_config.archive;
-    anyhow::ensure!(
-        !store_is_archive || client_is_archive,
-        "The node is configured as non-archival but is using database of an archival node."
-    );
-    if !store_is_archive && client_is_archive {
-        let mut update = hot.store_update();
-        update.set_ser(DBCol::BlockMisc, near_store::db::IS_ARCHIVE_KEY, &true)?;
-        update.commit()?;
+    if store_is_archive != near_config.client_config.archive {
+        if store_is_archive {
+            tracing::warn!(
+                target: "neard",
+                "Opening an archival database.  \
+                 Ignoring `archive` client configuration and forcing the node to run in archive mode."
+            );
+            near_config.client_config.archive = true;
+        } else {
+            let mut update = hot.store_update();
+            update.set_ser(DBCol::BlockMisc, near_store::db::IS_ARCHIVE_KEY, &true)?;
```

Comment:
should we also print a warning here, that we're marking the database as 'archival'?